packages/prediction-app/src/features/model-templates-overview/components/UserOptionsFields/fields/NullableField.tsx

```diff
@@ -28,9 +56,37 @@ export const NullableField: React.FC<NullableFieldProps> = ({
 }) => {
     const label = optionConfig.title || optionKey
     const nonNullType = optionConfig.anyOf!.find(typeOption => typeOption.type !== FIELD_TYPES.NULL)
+    const [validationError, setValidationError] = useState<string>('')
 
-    const inputType = nonNullType?.type === FIELD_TYPES.INTEGER || nonNullType?.type === FIELD_TYPES.NUMBER ? 'number' : 'text'
-    const step = nonNullType?.type === FIELD_TYPES.NUMBER ? 'any' : nonNullType?.type === FIELD_TYPES.INTEGER ? '1' : undefined
+    const isNumberField = nonNullType?.type === FIELD_TYPES.INTEGER || nonNullType?.type === FIELD_TYPES.NUMBER
+    const validationSchema = isNumberField ? createNullableNumberSchema(nonNullType!.type) : null
+
+    const validateValue = (value: string) => {
```

Comment:
Doesn't react-hook-form support field-level validations? I would think you could use `useController({ control, rules: ...})` to handle some of this logic? But its looks ok, so not blocking.